src/xss-common-qsort.h

```diff
@@ -261,6 +275,49 @@ X86_SIMD_SORT_INLINE arrsize_t partition_vec(type_t *l_store,
 
     return amount_ge_pivot;
 }
+/*
+ * Parition one YMM register based on the pivot and returns the
+ * number of elements that are greater than or equal to the pivot.
+ */
+template <typename vtype, typename type_t, typename reg_t = typename vtype::reg_t>
+X86_SIMD_SORT_INLINE arrsize_t partition_vec_avx2(type_t *l_store,
+                                             type_t *r_store,
+                                             const reg_t curr_vec,
+                                             const reg_t pivot_vec,
+                                             reg_t &smallest_vec,
+                                             reg_t &biggest_vec)
+{
+    /* which elements are larger than or equal to the pivot */
+    typename vtype::opmask_t ge_mask = vtype::ge(curr_vec, pivot_vec);
+
+    int32_t amount_ge_pivot = vtype::double_compressstore(
+            l_store, r_store, ge_mask, curr_vec);
+
+    smallest_vec = vtype::min(curr_vec, smallest_vec);
+    biggest_vec = vtype::max(curr_vec, biggest_vec);
+    
+    return amount_ge_pivot;
+}
+
+// Generic function dispatches to AVX2 or AVX512 code
+template <typename vtype, typename type_t, typename reg_t = typename vtype::reg_t>
+X86_SIMD_SORT_INLINE arrsize_t partition_vec(type_t *l_store,
```

Comment:
Is there a reason we need this extra function? If we define `vtype::double_compressstore` for all the `zmm_vector` classes, then we might not need two versions of `partition_vec` function, right?